share/config.fish

```diff
@@ -19,30 +19,32 @@ end
 
 if status --is-interactive
 	# Existance of string is a good pre-2.3.0 check. Could also check $FISH_VERSION in the future.
-
+	# This is a "launch", not an issue caused by autoloading during upgrades.
 	if not contains "string" (builtin -n)
-		set_color red;
-		echo You appear to be trying to run an old fish binary
-		echo with newer scripts installed into $__fish_datadir
-		echo This is an unsupported configuration.
+		# the string.fish message to `exec` will probably not help here, so this will that.
+		set -g __is_launched_without_string 1
+
 		set_color --bold
-		echo THINGS WILL NOT WORK.
+		echo "You appear to be trying to launch an old fish binary with newer scripts "
+		echo "installed into"; set_color --underline; echo -n "$__fish_datadir"
 		set_color normal
-		echo You may need to uninstall and reinstall fish.
-		sleep 10
-	end
-
-	# Enable truecolor/24-bit support for select terminals
-	if not set -q NVIM_LISTEN_ADDRESS # (Neovim will swallow the 24bit sequences, rendering text white)
-		and begin
-			set -q KONSOLE_PROFILE_NAME # KDE's konsole
-			or string match -q -- "*:*" $ITERM_SESSION_ID # Supporting versions of iTerm2 will include a colon here
-			or string match -q -- "st-*" $TERM # suckless' st
-			or test "$VTE_VERSION" -ge 3600 # Should be all gtk3-vte-based terms after version 3.6.0.0
-			or test "$COLORTERM" = truecolor -o "$COLORTERM" = 24bit # slang expects this
+		echo "This is an unsupported configuration."
+		set_color yellow
+		echo "You may need to uninstall and reinstall fish!"
+		set_color normal
+	else 
```

Comment:
Also, even if you make a no-op there instead of doing an `else` - the 24bit color test breaks on 2.2.0 and end up trying to use 24bit colors on terminals that can't handle them. Just seemed like the kinder fix.